In [17]:

import numpy as np
from numpy.linalg import  eig
import scipy.linalg as la

## menggunakan lib numpy

In [39]:
a = np.array([1, 3])
B = np.array([[1, -1],
              [6, 4]])
B

array([[ 1, -1],
       [ 6,  4]])

In [40]:
# ketika matrix B di kali dengan vactor a
B.dot(a)

array([-2, 18])

In [43]:
eigenvalues, eigenvector = eig(B)
print(eigenvalues.real)
print(eigenvector.real)

[2.5 2.5]
[[-0.23145502 -0.23145502]
 [ 0.9258201   0.9258201 ]]


## menggunakan lib scipy

In [44]:
result = la.eig(B)

In [45]:
val = result[0].real
val

array([2.5, 2.5])

In [48]:

vec = result[1].real
vec

array([[-0.23145502, -0.23145502],
       [ 0.9258201 ,  0.9258201 ]])

### pada matrix B memiliki 2 eigenvalues yaitu 2.5 dan 2.5, sedangkan memiliki 2 eigenvector [-0.23145502, 0.9258201] dan [-0.23145502,  0.9258201]

### jadi jawaban apakah vec a bukan eigen vactor dari matrix B, dan eigenvalues dari matrix B yaitu 2.5 dan 2.5

### coret coret

In [31]:
A = np.array([[3,2], [1, 0]])

values, vector = eig(A)

In [32]:
values

array([ 3.56155281, -0.56155281])

In [33]:
vector

array([[ 0.96276969, -0.48963374],
       [ 0.27032301,  0.87192821]])

In [36]:
e = np.array([0.96276969, 0.27032301])

A.dot(e)

array([3.42895509, 0.96276969])

In [37]:
e*3.56155281

array([3.42895509, 0.96276968])